stan/math/prim/prob/wiener5_lpdf.hpp

```diff
@@ -0,0 +1,708 @@
+            - newsign
+                  * exp(result - (log_density - error_term) - 2.5 * log_y_asq
+                        - 0.5 * LOG_TWO - 0.5 * LOG_PI));
+  } else {
+    ans = -(density_part_one
+            + newsign * exp(result - (log_density - error_term) + 2 * LOG_PI));
+  }
+  return WrtLog ? ans * exp(log_density) : ans;
+}
+
+/**
+ * Calculate the derivative of the wiener5 density w.r.t. 'sv'
+ *
+ * @tparam WrtLog Whether to return the derivative w.r.t.
+ *                  the natural (true) or log-scale (false) density
+ *
+ * @param y A scalar variable; the reaction time in seconds
+ * @param a The boundary separation
+ * @param v_value The drift rate
+ * @param w_value The relative starting point
+ * @param sv The inter-trial variability of the drift rate
+ * @param err The log error tolerance
+ * @return Gradient w.r.t. sv
+ */
+template <bool WrtLog = false>
+inline double wiener5_grad_sv(double y, double a, double v_value,
+                              double w_value, double sv,
+                              double err = log(1e-12)) noexcept {
+  const double one_plus_svsqr_y = 1 + square(sv) * y;
+  const double w = 1.0 - w_value;
+  const double v = -v_value;
+  const double t1 = -y / one_plus_svsqr_y;
+  const double t2 = (square(a * w) + 2 * a * v * w * y + square(v * y))
+                    / square(one_plus_svsqr_y);
+  const double ans = sv * (t1 + t2);
+  return WrtLog ? ans * wiener5_density<true>(y, a, v_value, w_value, sv, err)
+                : ans;
+}
+
+/**
+ * Utility function for replacing a value with a specified error value
+ */
+template <size_t NestedIndex>
+inline void assign_err(double arg, double err) {
+  arg = err;
+}
+
+/**
+ * Utility function for replacing a value with a specified error value,
+ * overload for use when the value is stored within a tuple.
+ */
+template <size_t NestedIndex, typename... TArgs>
+inline void assign_err(std::tuple<TArgs...>& args_tuple, double err) {
+  std::get<NestedIndex>(args_tuple) = err;
+}
+
+/**
+ * Utility function for estimating a function with a given set of arguments,
+ * checking the result against a provided error tolerance, and re-estimating
+ * the function with the increased error if it fails.
+ *
+ * @tparam ErrIndex Position of the error argument in the provided arguments
+ * @tparam NestedIndex Nested position if the error argument is within a tuple
+ * @tparam F Type of functor
+ * @tparam ArgsTupleT Type of tuple of arguments for functor
+ *
+ * @param functor Function to apply
+ * @param err Error value to check against
+ * @param args_tuple Tuple of arguments to pass to functor
+ * @param log_result Whether the function result is already on the log-scale
+ */
+template <size_t ErrIndex, bool GradW7 = false, size_t NestedIndex = 0,
+          bool LogResult = true, typename F, typename... ArgsTupleT>
+double estimate_with_err_check(const F& functor, double err,
+                               ArgsTupleT&&... args_tuple) {
```

Comment:
I think the last thing from here is that I'd like to get rid of `ErrIndex`. Can the functor's passed here just return a pair with the error in the pair? I think that makes things a lot simpler to reason about

Reply:
Ok. I deleted the ErrIndex. Since the error term is always located at the end of the parameter tuples, I replaced ErrIndex by the last tuple entry:
`size_tuple = std::tuple_size<decltype(err_args_tuple)>::value - 1;`

Reply:
I'm sorry. I was confused. My suggestion did not work because for the full model we fist have to extract the params-tuple before we can extract the error-term from it. In smaller models with no integration we can directly extract the error-term.